__tests__/send-tx.test.js

```diff
@@ -0,0 +1,121 @@
+      .set({ "x-wallet-id": TestUtils.walletId });
+    expect(response.status).toBe(400);
+    expect(response.body.succes).toBeFalsy();
+
+    response = await TestUtils.request
+      .post("/wallet/send-tx")
+      .send({
+        outputs: [{ value: 1 }],
+      })
+      .set({ "x-wallet-id": TestUtils.walletId });
+    expect(response.status).toBe(400);
+    expect(response.body.succes).toBeFalsy();
+  });
+
+  it("should receive an error when trying to do concurrent transactions (lock/unlock behavior)", async () => {
+    const promise1 = TestUtils.request
+      .post("/wallet/send-tx")
+      .send({
+        outputs: [{ address: "WPynsVhyU6nP7RSZAkqfijEutC88KgAyFc", value: 1 }],
+      })
+      .set({ "x-wallet-id": TestUtils.walletId });
+    const promise2 = TestUtils.request
+      .post("/wallet/send-tx")
+      .send({
+        outputs: [{ address: "WPynsVhyU6nP7RSZAkqfijEutC88KgAyFc", value: 1 }],
+      })
+      .set({ "x-wallet-id": TestUtils.walletId });
+
+    const [response1, response2] = await Promise.all([promise1, promise2]);
+    expect(response1.status).toBe(200);
+    expect(response1.body.hash).toBeTruthy();
+    expect(response2.status).toBe(200);
+    expect(response2.body.success).toBeFalsy();
+  });
+
+  it("should accept a custom token transaction", async () => {
+    const response = await TestUtils.request
+      .post("/wallet/send-tx")
+      .send({
+        outputs: [{ address: "WPynsVhyU6nP7RSZAkqfijEutC88KgAyFc", value: 1 }],
+        token: {
+          name: "stub_token",
+          uid: "09",
+          symbol: "stub_token",
+        },
+      })
+      .set({ "x-wallet-id": TestUtils.walletId });
+    expect(response.status).toBe(200);
+    expect(response.body.hash).toBeTruthy();
+  });
+
+  it("should not accept a custom token transaction without funds to cover it", async () => {
+    const response = await TestUtils.request
+      .post("/wallet/send-tx")
+      .send({
+        outputs: [{ address: "WPynsVhyU6nP7RSZAkqfijEutC88KgAyFc", value: 1 }],
+        token: {
+          name: "stub_token_2",
+          uid: "02",
+          symbol: "stub_token_2",
+        },
+      })
+      .set({ "x-wallet-id": TestUtils.walletId });
+    expect(response.status).toBe(200);
+    expect(response.body.success).toBeFalsy();
+  });
+
+  it("should not accept a custom token transaction without all token properties", async () => {
+    ['name', 'uid', 'symbol'].forEach(async (field) => {
+      const token = {
+        name: "stub_token",
+        uid: "01",
+        symbol: "stub_token",
+      };
+      delete token[field];
+      const response = await TestUtils.request
+        .post("/wallet/send-tx")
+        .send({
+          outputs: [{ address: "WPynsVhyU6nP7RSZAkqfijEutC88KgAyFc", value: 1 }],
+          token: token,
+        })
+        .set({ "x-wallet-id": TestUtils.walletId });
+      expect(response.status).toBe(400);
+      expect(response.body.success).toBeFalsy();
+    });
+  });
+});
```

Comment:
I think we should add more tests here:

1. Test using a specific input (and validate it really used it)
2. Test using an invalid change address
3. Test using a valid change address (and validate that the change output went to it)
4. Test invalid value, e.g. 0

Reply:
1. Test using a specific input (and validate it really used it)
Testing it with a mocked fullnode is not a valid effort for now.
2. Test using an invalid change address
Done
3. Test using a valid change address ~~(and validate that the change output went to it)~~
Done
4. Test invalid value, e.g. 0
Done